generator/src/main/java/com/scottlogic/deg/generator/constraints/StringConstraintsCollection.java

```diff
@@ -0,0 +1,180 @@
+
+        if (constraint instanceof NotConstraint){
+            AtomicConstraint complementOfConstraint = complementOfConstraint(((NotConstraint) constraint).negatedConstraint); // field1 < 10 -> field1 > 9
+            return complementOfConstraint != null
+                ? otherConstraints
+                    .stream()
+                    .anyMatch(otherConstraint -> isContradictory(complementOfConstraint, otherConstraint))
+                : false; //we cannot prove it is contradictory
+        }
+
+        return otherConstraints
+            .stream()
+            .anyMatch(otherConstraint -> isContradictory(constraint, otherConstraint));
+    }
+
+    private boolean isContradictory(AtomicConstraint constraint, AtomicConstraint otherConstraint) {
+        if (otherConstraint instanceof NotConstraint){
+            AtomicConstraint negatedConstraint = ((NotConstraint) otherConstraint).negatedConstraint;
+            if (constraint.equals(negatedConstraint)){
+                return true; //e.g field1 < 10, not(field1 < 10); //fully contradictory
+            }
+
+            AtomicConstraint complementOfOtherConstraint = complementOfConstraint(negatedConstraint); // field1 < 10 -> field1 > 9
+            return complementOfOtherConstraint != null
+                ? isContradictory(constraint, complementOfOtherConstraint)
+                : false; //we cannot prove it is contradictory
+        }
+
+        if (constraint.getClass().equals(otherConstraint.getClass())){
+            if (constraint instanceof StringHasLengthConstraint){
+                return areLengthConstraintsContradictory((StringHasLengthConstraint)constraint, (StringHasLengthConstraint)otherConstraint);
+            }
+
+            return false; //same type constraints cannot contradict
+        }
+
+        if (constraint instanceof IsStringShorterThanConstraint){
+            return isContradictoryToMaxLength(((IsStringShorterThanConstraint) constraint).referenceValue, otherConstraint)
+                .orElse(false);
+        }
+
+        if (constraint instanceof IsStringLongerThanConstraint){
+            return isContradictoryToMinLength(((IsStringLongerThanConstraint) constraint).referenceValue, otherConstraint)
+                .orElse(false);
+        }
+
+        if (constraint instanceof StringHasLengthConstraint) {
+            return isContradictoryToOfLengthConstraint(((StringHasLengthConstraint) constraint).referenceValue, otherConstraint)
+                .orElse(false);
+        }
+
+        return false; //we cannot prove it is contradictory
+    }
+
+    private boolean areLengthConstraintsContradictory(
+        StringHasLengthConstraint constraint,
+        StringHasLengthConstraint otherConstraint) {
+        return constraint.referenceValue != otherConstraint.referenceValue;
+    }
+
+    private Optional<Boolean> isContradictoryToMaxLength(int shorterThan, AtomicConstraint otherConstraint) {
```

Comment:
There is no need for this to be optional

Reply:
The method will emit:
* `true` = constraints contradict
* `false` = constraints do not contradict
* `Optional.empty` = constraints are not known to contradict, they could or could not.

So the method can return 3 states, hence the return type.

Reply:
Changed